homeassistant/components/device_tracker/actiontec.py

```diff
@@ -82,37 +99,56 @@ def __init__(self, config):
         # Test the router is accessible
         data = self.get_actiontec_data()
         self.success_init = data is not None
+        _LOGGER.info("actiontec scanner initialized")
+        if self.home_interval:
+            _LOGGER.info("home_interval set to: %s", self.home_interval)
 
     def scan_devices(self):
         """ Scans for new devices and return a
             list containing found device ids. """
 
         self._update_info()
-        return [client['mac'] for client in self.last_results]
+        return [client.mac for client in self.last_results]
 
     def get_device_name(self, device):
         """ Returns the name of the given device or None if we don't know. """
         if not self.last_results:
             return None
         for client in self.last_results:
-            if client['mac'] == device:
-                return client['ip']
+            if client.mac == device:
+                return client.ip
         return None
 
     @Throttle(MIN_TIME_BETWEEN_SCANS)
     def _update_info(self):
         """ Ensures the information from the Actiontec MI424WR router is up
             to date. Returns boolean if scanning successful. """
+        _LOGGER.info("Scanning")
         if not self.success_init:
             return False
 
         with self.lock:
-            # _LOGGER.info("Checking ARP")
-            data = self.get_actiontec_data()
-            if not data:
+            exclude_targets = set()
+            target_list = []
+            self.last_results = []
+            now = dt_util.now()
+            if self.home_interval:
+                for host in self.last_results:
+                    if host.last_update + self.home_interval > now:
+                        exclude_targets.add(host)
+                if len(exclude_targets) > 0:
+                    target_list = [t.ip for t in exclude_targets]
+
+            devices = self.get_actiontec_data()
+            if not devices:
                 return False
-            active_clients = [client for client in data.values()]
-            self.last_results = active_clients
+            for client in target_list:
+                if client in devices:

```

Comment:
In this for loop you have to  make sure to add the devices from previous scans that you want to keep back to last results. Now the code only makes sure they do not show up.